src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -177,7 +186,46 @@
      * @return An immutable map of subjects that have this permission set
      * @see SubjectCollection#getLoadedWithPermission(String)
      */
-    Map<Subject, Boolean> getAssignedSubjects(String collectionIdentifier);
+    Map<? extends Subject, Boolean> getAssignedSubjects(String collectionIdentifier);
+
+    /**
+     * Check if the given subject has the permission described, without any template parameters.
+     * If {@link #getId()} contains any template parameters, they will be stripped out -- See overloads if parameters are desired
+     *
+     * @param subj The subject to query
+     * @return Whether the given subject has this permission.
+     */
+    boolean query(Subject subj);
```

Comment:
I wonder whether it would be worth providing a default implementation of the PermissionDescription query methods - the functionality seems a bit complex - which is fine, but it's not overly clear how they should behave or be implemented - regex & string replacement I guess? Maybe provide a `Pattern` that can match the "template parameter" format?

Reply:
Queries on descriptions with template parameter should fail to execute (or always return false).
What would be your expected result? All in all, I think this method shouldn't exist:

> The description is meant to provide human readable *descriptions* and *meta data* for a permission. Descriptions are primarily *informational* [...]

So don't put any logic into them.

Reply:
I looked at doing a default implementation -- but that would involve exposing a "stripped id" to the api that isn't currently exposed.

Take a look at the [default implementation in SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/3106/files#diff-3dd1bd786a39a55be9ad739a27918645R89). I've put in some Patterns for a valid permission id and trailing template, but those could be pulled up to the api if you're interested.

---

ST, the reason for these permission description changes is to answer the question "Why should a plugin developer register their permissions?". With the existing system, there are very few benefits for plugins from these registrations, so nobody's going to use them. These helper methods and the default value method below aim to give plugin developers the ability to work entirely in permissions descriptions -- including basic parameterized permissions checks and setting metadata.